src/KvkClient.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Werkspot\KvkApi;
+
+use Werkspot\KvkApi\Client\KvkPaginator;
+use Werkspot\KvkApi\Client\KvkPaginatorInterface;
+use Werkspot\KvkApi\Client\Factory\KvkPaginatorFactoryInterface;
+use Werkspot\KvkApi\Http\Search\ProfileQuery;
+use Werkspot\KvkApi\Http\ClientInterface;
+use Werkspot\KvkApi\Http\Endpoint\MapperInterface;
+
+final class KvkClient
+{
+    /**
+     * @var ClientInterface
+     */
+    private $httpClient;
+
+    /**
+     * @var KvkPaginatorFactoryInterface
+     */
+    private $profileResponseFactory;
+
+    public function __construct(ClientInterface $httpClient, KvkPaginatorFactoryInterface $profileResponseFactory)
+    {
+        $this->httpClient = $httpClient;
+        $this->profileResponseFactory = $profileResponseFactory;
+    }
+
+    public function getProfile(ProfileQuery $profileQuery): KvkPaginator
+    {
+        $json = $this->httpClient->getJson($this->httpClient->getEndpoint(MapperInterface::PROFILE, $profileQuery));
+        $data = $this->decodeJsonToArray($json);
+
+        return $this->profileResponseFactory->fromProfileData($data);
+    }
+
+    public function getNextPage(KvkPaginatorInterface $kvkPaginator): KvkPaginatorInterface
+    {
+        $json = $this->httpClient->getJson($this->httpClient->getUrl($kvkPaginator->getNextUrl()));
+        $data = $this->decodeJsonToArray($json);
+
+        return $this->profileResponseFactory->fromProfileData($data);
+    }
+
+    public function getPreviousPage(KvkPaginatorInterface $kvkPaginator): KvkPaginatorInterface
+    {
+        $json = $this->httpClient->getJson($this->httpClient->getUrl($kvkPaginator->getPreviousUrl()));
+        $data = $this->decodeJsonToArray($json);
+
+        return $this->profileResponseFactory->fromProfileData($data);
+    }
+
+    /**
+     * @param $json
+     * @return mixed
+     */
+    private function decodeJsonToArray($json)
```

Comment:
type hint `string` as parameter type and then we can get rid of this docblock. the return type should be an array.